redis/commands/search/query.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, query_string: str) -> None:
         self._highlight_fields: List = []
         self._language: Optional[str] = None
         self._expander: Optional[str] = None
-        self._dialect: Optional[int] = None
+        self._dialect: int = 2
```

Comment:
I'd suggest creating a new file like `dialect.py` and defining this magic number as a constant there.